provision/kubernetes/helpers.go

```diff
@@ -121,11 +122,11 @@ func waitFor(timeout time.Duration, fn func() (bool, error), onTimeout func() er
 			return nil
 		}
 		select {
-		case <-timeoutCh:
-			if onTimeout == nil {
-				err = errors.Errorf("timeout after %v", timeout)
+		case <-ctx.Done():
+			if onCancel == nil {
+				err = errors.Errorf("canceled after %v", time.Since(start))
```

Comment:
I think we should call `err = errors.Wrapf(ctx.Err(), ...` here. This way we would be able to differentiate between DeadlineExceeded and explicit cancelation of events.